src/gui/curses/gui-curses-key.c

```diff
@@ -98,19 +98,24 @@ gui_key_default_bindings (int context)
         BIND(/* m-x           */ "meta-x",             "/input zoom_merged_buffer");
         BIND(/* m-d           */ "meta-d",             "/input delete_next_word");
         BIND(/* ^K            */ "ctrl-K",             "/input delete_end_of_line");
-        BIND(/* m-r           */ "meta-r",             "/input delete_line");
+        BIND(/* m-r           */ "meta-r",             "/input delete_input");
+        BIND(/* m-R           */ "meta-R",             "/input delete_line");
         BIND(/* ^T            */ "ctrl-T",             "/input transpose_chars");
         BIND(/* ^U            */ "ctrl-U",             "/input delete_beginning_of_line");
         BIND(/* ^Y            */ "ctrl-Y",             "/input clipboard_paste");
         BIND(/* <home>        */ "meta2-1~",           "/input move_beginning_of_line");
         BIND(/* <home>        */ "meta2-H",            "/input move_beginning_of_line");
         BIND(/* <home>        */ "meta2-7~",           "/input move_beginning_of_line");
         BIND(/* <home>        */ "meta-OH",            "/input move_beginning_of_line");
```

Comment:
I don't know what terminal emulator produces this escape sequence (`/* <home> */ "meta-OH"`), so I haven't added the equivalent shift-home sequence. The same goes for `/* <end> */ "meta-OF"`, `/* ^<up> */ "meta-OA"` and `/* ^<down> */ "meta-OB"`. The escape sequences that I have added are the ones produced by xterm and urxvt.

Reply:
Not relevant anymore with the new key bindings.